storm-server/src/main/java/org/apache/storm/scheduler/resource/User.java

```diff
@@ -79,15 +79,18 @@ public String getId() {
         return ret;
     }
 
-    public void markTopoUnsuccess(TopologyDetails topo, Cluster cluster) {
+    public void markTopoUnsuccess(TopologyDetails topo, Cluster cluster, String msg) {
         unsuccess.add(topo);
         if (cluster != null) {
-            cluster.setStatus(topo.getId(), "Scheduling Attempted but topology is invalid");
+            if (msg == null) {
+                msg = "Scheduling Attempted but topology is invalid";
```

Comment:
failed to schedule does not necessarily `the topology is invalid`.

Reply:
That message is a generic default - same as prior default. I believe there is one other caller to this method.